### DISCUSSION TITLE
Easiest Way to restart the tshock Server

### DISCUSSION THREAD

**Discussion by P-Faust** [Pryaxis/TShock#2394] [Q&A] [answered]
Hey there i'm new to Linux and Tshock. 
So if i change anything in the config file, i think i need to do an restart. 
My google research gave me /restart but that command is not nown if i type it in Terminal.
So my Question whats the easiest way to restart the server with given Permissions.
If i /save and close the terminal and start again with mono ./TerrariaServer.exe i need to re-enter the Permissions.
Is there an easier way to restart ? 
Thanks for the Help :)

**Comment by Kojirremer** [comment, marked as answer]
The /restart command was pretty useless, so it was deprecated a few months ago. Sure, it restarted the server, but didn't use any commandline parameters you may have used on the previous boot, so it would immediately get stuck waiting for user input for world selection, port, number of players, etc.

I'm assuming you want to have a fully automatic reload, in which case you need to start it via a batch file on windows, or its equivalent script on Linux, which restarts the server with your desired commandline parameters whenever you /exit it.

**Reply by P-Faust** [reply]
Ok thank you :) then i will later try to write an bash script that starts the server with the permissions i want :)

**Reply by bartico6** [reply]
You should use /reload for most config changes but in the case of you needing a hard restart you will need a batch script that sets the launch parameters as said above.

**Reply by P-Faust** [reply]
oh i tought it needs a hard restart after config changes :D Thank you that helps me alot. But i will try to write my first batch script for an hard reset, to learn batch scripting. Thank you for the fast answers 👍
